interpreter/function/builtin/digest_base64_decode.go

```diff
@@ -37,10 +41,63 @@ func Digest_base64_decode(ctx *context.Context, args ...value.Value) (value.Valu
 	}
 
 	input := value.Unwrap[*value.String](args[0])
-	dec, err := base64.StdEncoding.DecodeString(input.Value)
-	if err != nil {
-		return value.Null, err
+	removed := Digest_base64_decode_removeInvalidCharacters(input.Value)
+	dec, _ := base64.StdEncoding.DecodeString(removed)
+
+	return &value.String{Value: string(terminateNullByte(dec))}, nil
+}
+
+// Base64 decoding utility functions - they are also called by digest.base64url_decode and digest.base64_decode_nopad
+var nullByte = []byte{0}
+
+// Stop and return the point of Null-Byte found
+func terminateNullByte(decoded []byte) []byte {
+	if found := bytes.Index(decoded, nullByte); found != -1 {
+		return decoded[:found]
+	}
+	return decoded
+}
+
+// Even stopping decoding, we need to padding sign to success golang base64 decoding
+func base64_padding(b []byte) []byte {
+	for len(b)%4 > 0 {
+		b = append(b, 0x3D)
+	}
+	return b
+}
+
+func Digest_base64_decode_removeInvalidCharacters(input string) string {
+	removed := new(bytes.Buffer)
+	r := bufio.NewReader(strings.NewReader(input))
+
+	for {
```

Comment:
This question applies for the other `*removeInvalidCharacters` functions. Is there a specific reason why you went with the `bufio.Reader` approach here instead of ranging over the `input` string?

Reply:
I intend to use peek bytes to check the next byte without string range checking.